Add support for suite and case IDs

emanchado opened the pull request:
We are going to add an id to test suites. For that to happen, however,
we first need to _allow_ it without requiring it, then change kpet-db
to add it, then change the property to be required.

This is the first step.

Ref. FASTMOVING-1278

emanchado committed:
Accept "name" as an optional property in suites

We are going to add a name to test suites. For that to happen,
however, we first need to _allow_ it without requiring it, then change
kpet-db to add it, then change the property to be required. This is
the first step.

As part of this, expose the test suite name to Beaker XML templates so
that it can be exported in the kpet-db templates.

Ref. FASTMOVING-1278

spbnick reviewed (changes requested):
This is certainly required, but not sufficient. To start modifying kpet-db you need this exposed to the templates, and that wouldn't be done unless you modify `kpet.run.Suite.__init__()`. Consider adding an integration test outputting a suite ID to Beaker XML, as well.

spbnick committed:
data: Make suite name default to description

Make the suite names default to descriptions, so that we can start using
them before we have descriptions renamed to names in kpet-db.

spbnick committed:
data: Require either suite name or description

In preparation for transitioning from suite descriptions to suite names
make descriptions optional, but require either the name or the
description to be present.

This allows us to switch kpet-db to names in one go.

spbnick committed:
Add support for unique suite and case IDs

Add support for unique suite and case IDs. Make them optional until
kpet-db has them defined. Those IDs then could be used to identify
particular suites and cases throughout the CI stack, and in particular
in data warehouse and KCIDB.

spbnick committed:
Switch to using suite names

Switch to using (but not yet specifying) suite names instead of
descriptions, support for which is going to be removed.

wizardofoz123 commented on kpet/data.py:
```diff
@@ -314,17 +316,38 @@ def matches(self, target):
+        repeated_id_set = set()
+        for case in self.cases:
+            # pylint: disable=redefined-builtin, invalid-name
+            id = case.id
```
Shadows id()

spbnick replied:
Yeah, I knooow :(. I think naming a function "id" was a bad move by Python developers. Grabbing such a common name. That's why I told pylint to bugger off. However, if you'd prefer I didn't do this, I can rename them to "case_id" and "suite_id".

wizardofoz123 replied:
Unintentionally shadowing anything is something I never do. Maybe that's a habit from C.
If you really want to, you can merge the code as-is. I'll leave that up to you.

wizardofoz123 commented on kpet/data.py:
```diff
@@ -530,7 +560,23 @@ def get_sets_from_regex_list(regex_list):
-                                  format(suite.description, case.name))
+                                  format(suite.name, case.name))
+
+    def validate_suite_ids(self):
```
This code is basically a duplicate of validate_case_ids(). I'd probably like to see these merged.

spbnick replied:
Yeah, I thought about making a function returning a set of repeated members in a list, for example, but it wouldn't make this code much shorter. Do you still want me to get the common code out of these?

wizardofoz123 replied:
I'll leave that to you.

wizardofoz123 reviewed (changes requested):
If possible, please see and fix the 2 comments I made fixed. Otherwise LGTM, test against kpet-db and merge.

spbnick committed:
Add kpet.misc.get_repeated()

spbnick committed:
tests: Fix IntegrationMiscTests docstring
